src/main/java/com/hazelcast/aws/AwsEc2Api.java

```diff
@@ -65,6 +65,10 @@
         return parseDescribeInstances(response);
     }
 
+    boolean isUsePublicIp() {
```

Comment:
Instead of reaching to the configured value via this, `awsConfig.isUsePublicIp()` could have been a local field of `AwsEcsClient` and this would be also OK. However, since the public IPs are fetched through `AwsEc2Api#describeNetworkInterfaces` by `AwsEcsClient`, this check is kept here.